udata/api/oauth2.py

```diff
@@ -324,13 +335,15 @@ def query_client(client_id):
 
 def save_token(token, request):
     scope = token.pop("scope", "")
+    client = request.client
+    user = request.user or client.owner
     if request.grant_type == "refresh_token":
-        credential = request.credential
-        credential.update(scope=scope, **token)
+        old_token = OAuth2Token.objects(
+            refresh_token=request.refresh_token.refresh_token, client=client, user=user, scope=scope
+        ).first()
+        old_token.update(**token)
```

Comment:
Reading it again, should we return an HTTP error check if old_token is None?

Reply:
Slightly unrelated: I went back through the documentation, and saw [this](https://docs.authlib.org/en/latest/flask/2/grants.html#refresh-token-grant):
> By default, RefreshTokenGrant will not issue a refresh_token in the token response. Developers can change this behavior with:
>
> class RefreshTokenGrant(grants.RefreshTokenGrant):
>     INCLUDE_NEW_REFRESH_TOKEN = True

And I believe this is a good idea, so the refresh token is refreshed each and every time the token is refreshed.

More related to your point: I haven't seen anywhere in the code of the lib that would use the return of this function, and I feel reluctant to possibly change the behaviour.

What eases my mind is that we should never reach a state where we're trying to save a token that has been previously validated and accepted through `RefreshTokenGrant.authenticate_refresh_token`.

If my explanation has convinced you, let's land this, and if not, I'd be happy to dig further in this issue ;)